cli/kubectl_kadalu/logs.py

```diff
@@ -1,44 +1,106 @@
-    # TODO: make 'all-containers' optional based on container name (along with pod name).
     # TODO: provide options to pass options to kubectl logs (like tail, container name etc)
-    parser = subparsers.add_parser(name)
+
+    parser = subparsers.add_parser(
+        name,
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    arg = parser.add_argument
+
+    arg("-p", "--podname",
+        help="Specify pod name to get log info")
+
+    arg("-c", "--container",
+        help="Specify container name to get log info.\n"
+             "To be used along with '--podname'")
+
+    arg("-A", "--allcontainers", action="store_true",
+        help=("Show logs of all containers"
+              "of a particular pod.\n"
+              "To be used along with '--podname'"))
+
     utils.add_global_flags(parser)
 
 
-def validate(_args):
-    """No validation available"""
-    return
+def validate(args):
+    """Validate optional arguments"""
+
+    if args.container and not args.podname:
+        print("Specify pod name for the container '%s'." %args.container)
+        sys.exit(1)
+
+    if args.allcontainers and not args.podname:
+        print("Specify pod name for displaying of all containers.\n"
+              "Use without any optional arguments to get logs of "
+              "all pods and all of it's containers.")
+        sys.exit(1)
 
+    if args.allcontainers and args.container:
+        print("Cannot use both --allcontainers & --container together. "
+              "Choose any one of the above options.")
+        sys.exit(1)
+
+
+def fetch_pod_logs(args):
+    """ Get logs for a particular pod """
+    try:
+        # Set default to '--allcontainers' to avoid exception
+        container_name = ""
+        if args.allcontainers:
+            container_name = "--all-containers"
+        elif args.container:
+            container_name = args.container
+
+        log_cmd = utils.kubectl_cmd(args) + [
+            "logs", "-nkadalu",
+            args.podname, container_name]
+        log_resp = utils.execute(log_cmd)
+        print("----- (Kadalu Namespace) %s -----" % args.podname)
+        print(log_resp.stdout)
+        print()
+
+    except FileNotFoundError:
+        utils.kubectl_cmd_help(args.kubectl_cmd)
+    except utils.CommandError as err:
+        utils.command_error(log_cmd, err.stderr)
 
 def run(args):
     """ perform log subcommand """
 
     try:
-        cmd = utils.kubectl_cmd(args) + ["get", "pods", "-nkadalu", "-oname"]
-        resp = utils.execute(cmd)
-        pods = resp.stdout.split('\n')
-
-        for pod in pods:
-            if pod == "":
-                continue
-            log_cmd = utils.kubectl_cmd(args) + ["logs", "-nkadalu", pod, "--all-containers=true"]
-            log_resp = utils.execute(log_cmd)
-            print("----- (Kadalu Namespace) %s -----" % pod)
-            print(log_resp.stdout)
-            print()
+        if args.podname:
+            fetch_pod_logs(args)
```

Comment:
Just give a thought whether this function can be removed or not, as final commands in `if` and `else` are almost similar.

Reply:
I'll try to refactor since the print is same except for 'for' and 'gettting all pods'